src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -387,11 +472,17 @@ protected void getChanges2(ChangeEventSourceContext context,
                             YbProtoReplicationMessage message = new YbProtoReplicationMessage(
                                     m, this.yugabyteDBTypeRegistry);
 
+                            // Ignore safepoint record as they are not meant to be processed here.
+                            if (m.getOp() == Op.SAFEPOINT) {
```

Comment:
Ignoring `SAFEPOINT` records in `YugabyteDBStreamingChangeEventSource` i.e. for scenarios where consistency is not enabled.